### PR TITLE
Move crds to templates folder

### PR DESCRIPTION
**Pull request by areina**
Currently, the CRDs are created during `helm install` but they aren't updated. 

This PR moves them from `crds` to `templates` to solve that. 
It also introduces a new parameter to control if the `qdrantversions` CRD has to be installed.

### CODE REVIEW

**Review by pedjak — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -20,8 +20,9 @@ gen: manifests generate format vet ## Generate code containing DeepCopy, DeepCop
 
 .PHONY: manifests
 manifests: controller-gen ## Generate CustomResourceDefinition objects.
-	$(CONTROLLER_GEN) crd paths="./..." output:crd:artifacts:config=charts/qdrant-operator-crds/crds
-	cp charts/qdrant-operator-crds/crds/qdrant.io_qdrantversions.yaml charts/qdrant-operator-crds/crds-for-cluster-api/qdrant.io_qdrantversions.yaml
+	$(CONTROLLER_GEN) crd paths="./..." output:crd:artifacts:config=charts/qdrant-operator-crds/templates/crds
+	sed -i '' '1s/^/{{ if .Values.includeQdrantVersionsCRD }}\n/'  charts/qdrant-operator-crds/templates/crds/qdrant.io_qdrantversions.yaml
```
[the remark this answers is not in the document]
Comment (pedjak) [reply Makefile:24]: What component is consuming/creating instances of QdrantVersion?  Perhaps we should add this to the chart of this component? CRD do not need to be part of this repo, they could be rendered in projects where they are needed.

[the remark this answers is not in the document]
Comment (pedjak) [reply Makefile:24]: I would actually propose to drop all CRD yaml files from this repo too and only have them available through charts.

File: Makefile [hunk 2]
```diff
@@ -20,8 +20,20 @@ gen: manifests generate format vet ## Generate code containing DeepCopy, DeepCop
 
 .PHONY: manifests
 manifests: controller-gen ## Generate CustomResourceDefinition objects.
-	$(CONTROLLER_GEN) crd paths="./..." output:crd:artifacts:config=charts/qdrant-operator-crds/crds
-	cp charts/qdrant-operator-crds/crds/qdrant.io_qdrantversions.yaml charts/qdrant-operator-crds/crds-for-cluster-api/qdrant.io_qdrantversions.yaml
+	rm charts/qdrant-operator-crds/templates/management-crds/*.yaml
+	rm charts/qdrant-operator-crds/templates/region-crds/*.yaml
+	$(CONTROLLER_GEN) crd paths="./..." output:crd:artifacts:config=charts/qdrant-operator-crds/templates
+	mv charts/qdrant-operator-crds/templates/qdrant.io_qdrantversions.yaml charts/qdrant-operator-crds/templates/management-crds/
```
Comment (Robert-Stam) [inline Makefile:26]: Maybe not for this PR, but we want to rename QdrantVersion to QdrantRelease

Comment (pedjak) [reply Makefile:26]: yup, there is [ticket](https://github.com/qdrant/cloud-pm/issues/1539) for that.

File: Makefile [hunk 3]
```diff
@@ -20,8 +20,20 @@ gen: manifests generate format vet ## Generate code containing DeepCopy, DeepCop
 
 .PHONY: manifests
 manifests: controller-gen ## Generate CustomResourceDefinition objects.
-	$(CONTROLLER_GEN) crd paths="./..." output:crd:artifacts:config=charts/qdrant-operator-crds/crds
-	cp charts/qdrant-operator-crds/crds/qdrant.io_qdrantversions.yaml charts/qdrant-operator-crds/crds-for-cluster-api/qdrant.io_qdrantversions.yaml
+	rm charts/qdrant-operator-crds/templates/management-crds/*.yaml
```
Comment (pedjak) [inline Makefile:23]: do we need to keep them in git repo at all, if everybody just consumes the published chart?

File: Makefile [hunk 4]
```diff
@@ -20,8 +20,20 @@ gen: manifests generate format vet ## Generate code containing DeepCopy, DeepCop
 
 .PHONY: manifests
 manifests: controller-gen ## Generate CustomResourceDefinition objects.
-	$(CONTROLLER_GEN) crd paths="./..." output:crd:artifacts:config=charts/qdrant-operator-crds/crds
```
Comment (pedjak) [inline Makefile:23]: these two lines got expanded into 13 lines. I would suggest to move this logic into a script under `hack` directory, keeping the makefile easier to read and making the script testable.

**Pull request merged**